### ISSUE TITLE
[BUG] FatalProspectorException: Tool pylint failed to run

### ISSUE THREAD

**Issue by sergluka** [PyCQA/prospector#575]
**Describe the bug** 

Running Prospector cause an exception with the message: "Tool pylint failed to run"

```
$ prospector -o xunit:code-analysis-junit.xml -X
Traceback (most recent call last):
  File "/usr/py-3.9/lib/python3.9/site-packages/prospector/run.py", line 87, in execute
    messages += tool.run(found_files)
  File "/usr/py-3.9/lib/python3.9/site-packages/prospector/tools/pylint/__init__.py", line 239, in run
    self._linter.check(self._args)
  File "/usr/py-3.9/lib/python3.9/site-packages/pylint/lint/pylinter.py", line 696, in check
    ast_per_fileitem = self._get_asts(fileitems, data)
  File "/usr/py-3.9/lib/python3.9/site-packages/pylint/lint/pylinter.py", line 707, in _get_asts
    for fileitem in fileitems:
  File "/usr/py-3.9/lib/python3.9/site-packages/pylint/lint/pylinter.py", line 874, in _iterate_file_descrs
    for descr in self._expand_files(files_or_modules).values():
  File "/usr/py-3.9/lib/python3.9/site-packages/prospector/tools/pylint/linter.py", line 36, in _expand_files
    if not self._files.is_excluded(Path(module["path"])):
TypeError: string indices must be integers
The above exception was the direct cause of the following exception:
Traceback (most recent call last):
  File "/usr/py-3.9/bin/prospector", line 8, in <module>
    sys.exit(main())
  File "/usr/py-3.9/lib/python3.9/site-packages/prospector/run.py", line 199, in main
    prospector.execute()
  File "/usr/py-3.9/lib/python3.9/site-packages/prospector/run.py", line 105, in execute
    raise FatalProspectorException(f"Tool {toolname} failed to run.") from ex
prospector.exceptions.FatalProspectorException: Tool pylint failed to run.
```

**Environment**

- OS: Amazon Linux via the Docker
- Tool: pylint
- Prospector version: 1.8.3
- Python version: 3.9.16

**Dependencies**

CacheControl-0.12.11 Flask-1.1.2 GitPython-3.1.30 Jinja2-3.1.2 MarkupSafe-2.1.1 PySocks-1.7.1 PyYAML-6.0 SecretStorage-3.3.3 aiocache-0.11.1 aiohttp-3.8.3 aiopg-1.0.0 aiosignal-1.3.1 astroid-2.13.2 async-generator-1.10 async-timeout-4.0.2 asyncpg-0.21.0 attrs-22.2.0 bandit-1.7.4 bleach-5.0.1 build-0.9.0 cachetools-5.2.1 certifi-2022.12.7 cffi-1.15.1 charset-normalizer-2.1.1 click-8.1.3 commonmark-0.9.1 contourpy-1.0.6 cryptography-39.0.0 cycler-0.11.0 cyclonedx-python-lib-3.1.3 cytoolz-0.12.1 dacite-1.7.0 dataclasses-0.6 deprecation-2.1.0 dill-0.3.6 docutils-0.19 dodgy-0.2.1 eth-hash-0.5.1 eth-typing-3.2.0 eth-utils-2.1.0 exceptiongroup-1.1.0 exchanges-2.3.6 exchanges-wrapper-0.0.0.dev0 fastlogging-0.9.6 flake8-5.0.4 flake8-polyfill-1.0.2 fonttools-4.38.0 frozenlist-1.3.3 gitdb-4.0.10 google-api-core-2.11.0 google-api-python-client-2.71.0 google-auth-2.15.0 google-auth-httplib2-0.1.0 google-auth-oauthlib-0.8.0 googleapis-common-protos-1.57.1 greenlet-2.0.1 guppy3-3.1.2 h11-0.14.0 html5lib-1.1 httplib2-0.21.0 idna-3.4 importlib-metadata-6.0.0 iniconfig-2.0.0 isort-5.11.4 itsdangerous-2.1.2 jaraco.classes-3.2.3 jeepney-0.8.0 keyring-23.13.1 kiwisolver-1.4.4 lazy-object-proxy-1.9.0 libmagic-1.0 lockfile-0.12.2 matplotlib-3.6.2 mccabe-0.7.0 more-itertools-9.0.0 msgpack-1.0.4 multidict-6.0.4 mypy-0.991 mypy-extensions-0.4.3 numpy-1.24.1 oauthlib-3.2.2 outcome-1.2.0 packageurl-python-0.10.4 packaging-21.3 pbr-5.11.0 pep517-0.13.0 pep8-naming-0.10.0 pillow-9.4.0 pip-api-0.0.30 pip-audit-2.4.12 pip-requirements-parser-32.0.1 pkginfo-1.9.6 platformdirs-2.6.2 pluggy-1.0.0 poetry-semver-0.1.0 prospector-1.8.3 protobuf-4.21.12 psutil-5.9.4 psycopg2-binary-2.9.5 pyasn1-0.4.8 pyasn1-modules-0.2.8 pycodestyle-2.9.1 pycparser-2.21 pycryptodome-3.16.0 pydocstyle-6.2.3 pyflakes-2.5.0 pygments-2.14.0 pyjwt-2.6.0 pylint-2.15.10 pylint-celery-0.3 pylint-django-2.5.3 pylint-flask-0.6 pylint-forbidden-imports-1.0.0 pylint-plugin-utils-0.7 pyotp-2.4.0 pyparsing-3.0.9 pyroma-4.1 pytest-7.2.0 pytest-asyncio-0.20.3 pytest-mock-3.10.0 pytest-rerunfailures-10.3 python-dateutil-2.8.2 python-engineio-3.14.2 python-gnupg-0.5.0 python-socketio-4.6.1 pyzmq-22.3.0 readme-renderer-37.3 requests-2.28.1 requests-oauthlib-1.3.1 requests-toolbelt-0.10.1 requirements-detector-1.0.3 resolvelib-0.9.0 rfc3986-2.0.0 rich-13.0.1 rsa-4.9 scipy-1.10.0 selenium-4.7.2 setoptconf-tmp-0.3.1 setuptools-65.6.3 six-1.16.0 smmap-5.0.0 sniffio-1.3.0 snowballstemmer-2.2.0 sortedcontainers-2.4.0 sqlalchemy-1.4.46 stevedore-4.1.1 toml-0.10.2 tomli-2.0.1 tomlkit-0.11.6 toolz-0.12.0 trio-0.22.0 trio-websocket-0.9.2 trove-classifiers-2022.12.22 twine-4.0.2 types-mock-5.0.0.0 typing-extensions-4.4.0 ujson-5.5.0 undetected-chromedriver-3.2.1 uritemplate-4.1.1 urllib3-1.26.13 uvloop-0.15.3 vulture-2.7 webencodings-0.5.1 websockets-10.4 werkzeug-2.2.2 wheel-0.38.4 wrapt-1.14.1 wsproto-1.2.0 yarl-1.8.2 zipp-3.11.0

Config: [prospector.yaml.zip](https://github.com/PyCQA/prospector/files/10373617/prospector.yaml.zip)

**Comment by christokur** [comment]
This is the cause for the failure:
 https://github.com/PyCQA/prospector/blob/99be9b8ddc705c38fffd12d82a11812ef4e9d0ef/prospector/tools/pylint/linter.py#L27

```
False == '2.15.10' > '2.5.6' 

```

Cannot do version comparison as a string ...

**Comment by Pierre-Sassoulas** [comment]
@christokur do you want to contribute the fix now that you investigated the issue ?

**Comment by christokur** [comment]
I will make a patch provided I don't have to hunt for `packaging` ;)

**Comment by ryancausey** [comment]
Super awesome to see a fix to the issue I just ran into already merged. Thanks to everyone!

**Comment by sergluka** [comment]
Fixed in v1.8.4
